golos.ctrl/src/golos.ctrl.cpp

```diff
@@ -177,25 +177,35 @@ void control::unregwitness(name witness) {
     assert_started();
     require_auth(witness);
 
-    // TODO: simplify upsert to allow passing just inner lambda
-    bool exists = upsert_tbl<witness_tbl>(witness, [&](bool) {
-        return [&](witness_info& w) {
-            eosio_assert(w.active, "witness already unregistered");
-            w.active = false;
-        };
-    }, false);
-    eosio_assert(exists, "witness not found");
+    witness_tbl witness_table(_self, _self.value);
+    auto it = witness_table.find(witness.value);
+    witness_table.erase(*it);
```

Comment:
This inconsistent with `unvotewitness` action. How users can get vote back, in case witness unregistered yourself? It's better allow witness unreg action only if it doesn't have votes from users. _Notice: you need implement vote-counter in witness for this logic._